Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class NumericSpinEditor : NSView
+	{
+		protected NumericTextField numericEditor;
+		protected NSStepper stepper;
+		protected bool editing;
+
+
+		public event EventHandler ValueChanged;
+		public event EventHandler StepperValueChanged;
+		public event EventHandler FieldEditingEnded;
+
+		public ValidationType NumericMode {
+			get { return numericEditor.NumericMode; }
+			set {
+				numericEditor.NumericMode = value;
+				Reset ();
+			}
+		}
+
+		public string StringValue {
+			get { return numericEditor.StringValue; }
+			set { numericEditor.StringValue = value; }
+		}
+
+		public string PlaceholderString {
+			get { return ((NSTextFieldCell)numericEditor.Cell).PlaceholderString; }
+			set { ((NSTextFieldCell)numericEditor.Cell).PlaceholderString = value; }
+		}
+
+		public NSStepper Stepper {
+			get { return stepper; }
+		}
+
+		public override CGSize IntrinsicContentSize {
+			get {
+				var baseSize = stepper.IntrinsicContentSize;
+				return new CGSize (baseSize.Width + 20, baseSize.Height);
+			}
+		}
+
+		public NSColor BackgroundColor {
+			get {
+				return numericEditor.BackgroundColor;
+			}
+			set {
+				numericEditor.BackgroundColor = value;
+			}
+		}
+
+		public override nfloat BaselineOffsetFromBottom {
+			get { return numericEditor.BaselineOffsetFromBottom; }
+		}
+
+		public int Digits {
+			get { return (int)formatter.MaximumFractionDigits; }
+			set { formatter.MaximumFractionDigits = value; }
+		}
+
+		public virtual double Value {
+			get { return stepper.DoubleValue; }
```

Comment:
@alanmcgovern why worrying? Do you think the only access should be via StringValue?